pkg/kademlia/dialer.go

```diff
@@ -86,6 +88,27 @@ func (dialer *Dialer) Ping(ctx context.Context, target pb.Node) (bool, error) {
 	return err == nil, errs.Combine(err, conn.disconnect())
 }
 
+// GetPeerID pings target.
+func (dialer *Dialer) GetPeerID(ctx context.Context, target pb.Node) (pID *identity.PeerIdentity, err error) {
```

Comment:
I think this should have a name to indicate that it might connect to another system. `FetchPeerIdentity` or something? `ID` usually refers to the `NodeID` not the full/peer `Identity`